CONTRIBUTING.md

```diff
@@ -1,77 +1,150 @@
+
+#### Engagement
+Engagement includes feedback about learning materials and resources. It is always helpful to know where you prefer to be engaged and how you prefer to learn. 
+- "I would like to hear more about `<`specific topic`>` in `<`resource`>` so that I can achieve `<`goal`>`."
+- "`<`Resource`>` could be more inclusive and accessible if you considered `<`alternative`>` because `<`benefit`>`. Have you considered making these changes?"
+   
+Many of the resources we provide are also open source themselves! It can sometimes be more effective to leave feedback on the more specific repo. This includes:
+- [documentation](https://github.com/MicrosoftDocs/windows-dev-docs)
+- [website (`gh-pages` branch of this repo)](https://github.com/microsoft/microsoft-ui-xaml/tree/gh-pages)
+- [Xaml Controls Gallery (sample app)](https://github.com/microsoft/Xaml-Controls-Gallery/tree/master)
+
+#### Team Process
+Team process includes feedback about our repo and overall communication. Suggestions for improvement and proposed alternatives are very helpful here. 
+- Ex: "`<`specific aspect of team process - e.g., response frequency, format`>` of the repo is making it hard for me to achieve `<`goal`>`. Please consider doing `<`alternative`>` instead because that would help `<`action`>`" 
+
+## Code Contribution Guidelines
+Note this repository is not ready for open source collaboration at this time; this work is currently in progress. You can track the WinUI team's progress towards open source collaboration in the [Phased Rollout to Open Source Collaboration](https://github.com/microsoft/microsoft-ui-xaml/discussions/10700) discussion.
+
+WinUI will be taking a phased approach to opening up the repo:
+**Phase 1: Increased Mirror Frequency**
+After the WASDK 1.8 release (end of August), we'll begin more frequent mirroring of internal commits to GitHub to increase transparency and show progress.
+**Phase 2: 3rd Party Devs Build Locally**
+External developers will be able to clone and build the repo locally, with documentation to guide setup and dependencies.
+**Phase 3: 3rd Party Devs Contribute & Run Tests**
+Contributors will be able to submit PRs and run tests locally. We're working to untangle private dependencies and make test infrastructure publicly accessible.
+**Phase 4: GitHub as Center of Gravity**
+GitHub becomes the primary place for development, issue tracking, and community engagement. Internal mirrors will be phased out.
+
+### [WIP] New Contributors
+Contributions from the community are greatly appreciated. We mark the most straightforward issues with labels. These issues are the best place to start if you are interested in contributing but are new to the codebase.
 
+- [good first issues](https://github.com/Microsoft/microsoft-ui-xaml/labels/good first issue)
```

Comment:
For "good first issues", GitHub diff isn't correctly showing it as a link, perhaps due to spaces in the label name in the link itself. Perhaps those spaces need to be changed to "%20"s. Same for "feature request" link below.

Reply:
Same for the "auto merge" link at the bottom.